### PR TITLE
Cache yarn, test against latest node version

### PR DESCRIPTION
**Pull request by amilajack**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,8 @@ services: mongodb
 sudo: required
 node_js:
   - "8"
-  - "10"
+  - node
+cache: yarn
```
Comment (OmgImAlexis) [inline .travis.yml]: Please remove this, I've had nothing but issues with the yarn cache and it really doesn't speed anything up. We could switch to npm's ci mode if we really wanted faster installs.

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,8 @@ services: mongodb
 sudo: required
 node_js:
   - "8"
-  - "10"
+  - node
```
Comment (simison) [inline .travis.yml]: If we're to add "latest", I'd rather have it in "allowed failures" not to have CI surprises slowing us down later on (kinda what we have now with MongoDB installs failing in CI).

Comment (OmgImAlexis) [reply .travis.yml]: So maybe `LTS` and `latest`?

Comment (simison) [reply .travis.yml]: I prefer clearly declared and stable CI but if you prefer to have `LTS` here, I'm not going to block it. :-)

If someday we're not maintaining this actively enough, the day when Agenda becomes incompatible with LTS, all CI build would start failing all of a sudden.

Comment (OmgImAlexis) [reply .travis.yml]: My hope was once the ci build start failing that means we need to start migrating to the newer version, hopefully we're already there before it happens.

We could also do specific versions + latest instead?

Comment (simison) [reply .travis.yml]: Yep, so it's mostly a taste-preference:
- don't declare version compatibility to "un-foreseen future": CI informs us when we're behind
- declare strict upper compatibility limit: danger to stay behind without CI informing us


Considering that our codebase is pretty cutting edge right now, I don't see huge risk doing either way. 👍  I'll leave it to you and others to consider. :-)

Comment (simison) [reply .travis.yml]: >We could also do specific versions + latest instead?

I like this :thumbsup:

Comment (simison) [reply .travis.yml]: Noting also that ~if~ when CI breaks, we can "fix" it by simply moving LTS to "allowed failures": we don't necessarily have to scramble to build support to get CI back green.

Comment (OmgImAlexis) [reply .travis.yml]: You have me vote either way, so up to you.

**Comment by MichielDeMey** [comment]
We’ve deprecated Node 8 support in `v3.0.0` and are now testing against the latest Node versions. 👍

**Pull request closed without merging**